ledger-tool/src/main.rs

```diff
@@ -2766,6 +2912,93 @@ fn main() {
+                                                .map(|ix| {
+                                                    let program_id = accounts
+                                                        [ix.program_id_index as usize]
+                                                        .clone();
+
+                                                    let accounts = ix
+                                                        .accounts
+                                                        .iter()
+                                                        .map(|idx| accounts[*idx as usize].clone())
+                                                        .collect();
+
+                                                    let data = hex::encode(&ix.data);
+
+                                                    Instruction {
+                                                        program_id,
+                                                        accounts,
+                                                        data,
+                                                    }
+                                                })
+                                                .collect();
+
+                                            let execution_results =
+                                                batch.execution_results[no].clone();
+
+                                            let is_simple_vote_tx = tx.is_simple_vote_transaction();
+
+                                            Transaction {
+                                                accounts,
+                                                instructions,
+                                                is_simple_vote_tx,
+                                                execution_results,
+                                            }
+                                        })
+                                        .collect();
+
+                                    let batch = TransactionBatch {
+                                        bank_hash: batch.bank_hash.unwrap().to_string(),
+                                        transactions,
+                                    };
+
+                                    if let Some(recorded_slot) =
+                                        recorded_slots.iter_mut().find(|f| f.slot == slot)
+                                    {
+                                        recorded_slot.transactions.push(batch);
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    // writing the json file ends up with a syscall for each number, comma, indentation etc.
+                    // use BufWriter to speed things up
+
+                    let writer = std::io::BufWriter::new(recorded_slots_file);
+
+                    serde_json::to_writer_pretty(writer, &recorded_slots.unwrap()).unwrap();
```

Comment:
While this is also json, I'd like to re-use the existing struct (and thus file format) if possible. For example, I think it is nice that the existing bank hash details file spits out the client version which we lose here

Reply:
Makes sense, now we write `BankHashDetails`